ellipticoind/src/consensus.rs

```diff
@@ -0,0 +1,104 @@
+use crate::system_contracts::ellipticoin::Miner;
+use std::collections::HashMap;
+use serde_cose::Sign1;
+use ellipticoin::{BurnProofs, BurnTransaction, PublicKey, WitnessedMinerBlock, Address};
+use crate::config::{
+    my_public_key,
+    my_signing_key
+};
+
+use crate::constants::MINERS;
+use std::convert::TryFrom;
+
+
+#[derive(Clone, Debug)]
+pub enum MinerBlockDecision {
+    Burned(BurnProofs),
+    Accepted(WitnessedMinerBlock)
+}
+
+#[derive(Clone, Debug)]
+pub struct ExpectedBlock {
+    pub number: i32,
+    pub miner: Miner,
+    pub decisions: HashMap<PublicKey, MinerBlockDecision>,
+    pub burned_miners: HashMap<PublicKey, BurnProofs>,
```

Comment:
Higher level thing: Would it be possible to burn miners one at a time? Might simplify things.

Reply:
I'm not sure. I think different miners will be at different stages of the burn logic, and even after you have decided that a miner is burned, other miners may request the proof.